From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Samuel Attard <samuel.r.attard@gmail.com>
Date: Thu, 25 Jun 2020 09:29:04 -0700
Subject: fix: do not register the ESM loader in renderer processes

Only one ESM loader can be registered per isolate, in renderer processes this should be blink.  This patches node so that it won't register it's handler (overriding blinks) in non-browser processes.

diff --git a/lib/internal/bootstrap/pre_execution.js b/lib/internal/bootstrap/pre_execution.js
index b4a0f71af5853f427a10449b52509052fbe3facd..06297a4c258e2f0b715da35f855c788d776fc437 100644
--- a/lib/internal/bootstrap/pre_execution.js
+++ b/lib/internal/bootstrap/pre_execution.js
@@ -408,6 +408,8 @@ function initializeCJSLoader() {
 }
 
 function initializeESMLoader() {
+  // Do not hook the ESM loader in renderer processes as it overrides blinks loader
+  if (process.type !== 'browser') return;
   // Create this WeakMap in js-land because V8 has no C++ API for WeakMap.
   internalBinding('module_wrap').callbackMap = new SafeWeakMap();
 
